src/main/java/org/bihealth/mi/easysmpc/components/EntryEMailDetails.java

```diff
@@ -0,0 +1,246 @@
+                                                                    }
+                                                                }, false, false);
+        this.add(portEntry);
+        
+        // Encryption entry
+        radioEncryptionType = new ComponentRadioEntry(Resources.getString("EmailConfig.20"),
+                                                      Resources.getString("EmailConfig.21"),
+                                                      Resources.getString("EmailConfig.22"));
+        //radioEncryptionType.setChangeListener(this);
+        this.add(radioEncryptionType);
+    }
+
+    /**
+     * Creates a new instance. If settings is not null fields are pre-filled from it
+     * 
+     * @param title
+     * @param defaultPortImap
+     * @param settings
+     * @param isIMAP
+     */
+    public EntryEMailDetails(String title, int defaultPort, ConnectionIMAPSettings settings, boolean isIMAP) {
+        this(title, defaultPort);       
+        
+        // Check
+        if(settings == null) {
+            return;
+        }
+        
+        // Set values
+        if (isIMAP) {
+            this.serverEntry.setValue(settings.getIMAPServer());
+            this.portEntry.setValue(Integer.toString(settings.getIMAPPort()));
+            this.radioEncryptionType.setFirstOptionSelected(settings.isSSLTLSIMAP());
+        } else {
+            this.serverEntry.setValue(settings.getSMTPServer());
+            this.portEntry.setValue(Integer.toString(settings.getSMTPPort()));
+            this.radioEncryptionType.setFirstOptionSelected(settings.isSSLTLSSMTP());
+        }
+    }    
+
+    /**
+     * Creates a new instance. If oldDetails is not null fields are pre-filled from it
+     * 
+     * @param title
+     * @param defaultPort
+     * @param oldDetails
+     */
+    public EntryEMailDetails(String title, int defaultPort, EntryEMailDetails oldDetails) {
+        this(title, defaultPort);
+        
+        // Check
+        if(oldDetails == null) {
+            return;
+        }
+        
+        // Set values
+        this.serverEntry.setValue(oldDetails.getServerEntry().getValue());
+        this.portEntry.setValue(oldDetails.getPortEntry().getValue());
+        this.radioEncryptionType.setFirstOptionSelected(oldDetails.getRadioEncryptionType().isFirstOptionSelected());
+    }
+
+    /**
+     * @return the serverEntry
+     */
+    protected ComponentEntryOne getServerEntry() {
+        return serverEntry;
+    }
+
+    /**
+     * @return the portEntry
+     */
+    protected ComponentEntryOne getPortEntry() {
+        return portEntry;
+    }
+
+    /**
+     * @return the radioEncryption
+     */
+    protected ComponentRadioEntry getRadioEncryptionType() {
+        return radioEncryptionType;
+    }
+    
+    /**
+     * Sets a change listener
+     * @param listener
+     */
+    public void setChangeListener(ChangeListener listener) {
```

Comment:
Really needed to register external listeners?

Reply:
Yes, this is in line with the code pre-exsiting before this PR.